src/js/components/FileInput/FileInput.js

```diff
@@ -0,0 +1,246 @@
+import React, { forwardRef, useContext, useRef } from 'react';
+import styled, { ThemeContext } from 'styled-components';
+
+import { defaultProps } from '../../default-props';
+
+import { focusStyle, useForwardedRef } from '../../utils';
+
+import { Anchor } from '../Anchor';
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { FormContext } from '../Form/FormContext';
+import { Keyboard } from '../Keyboard';
+import { Stack } from '../Stack';
+import { Text } from '../Text';
+
+import { StyledFileInput } from './StyledFileInput';
+
+const ControlBox = styled(Box)`
+  cursor: pointer;
+  &:focus {
+    ${focusStyle}
+  }
+  ${props => props.theme.fileInput && props.theme.fileInput.extend};
+  ${props =>
+    props.hover &&
+    props.theme.fileInput &&
+    props.theme.fileInput.hover &&
+    props.theme.fileInput.hover.extend};
+  ${props =>
+    props.dragOver &&
+    props.theme.fileInput &&
+    props.theme.fileInput.dragOver &&
+    props.theme.fileInput.dragOver.extend};
+`;
+
+const Label = styled(Text)`
+  ${props =>
+    props.theme.fileInput &&
+    props.theme.fileInput.label &&
+    props.theme.fileInput.label.extend};
+`;
+
+const Message = styled(Text)`
+  ${props =>
+    props.theme.fileInput &&
+    props.theme.fileInput.message &&
+    props.theme.fileInput.message.extend};
+`;
+
+const FileInput = forwardRef(
+  (
+    {
+      a11yTitle,
+      background,
+      border,
+      id,
+      fileLabel,
+      messages,
+      margin,
+      multiple,
+      name,
+      onChange,
+      pad,
+      value: valueProp,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext);
+    const formContext = useContext(FormContext);
+    const [files, setFiles] = formContext.useFormInput(name, valueProp, []);
+    const [hover, setHover] = React.useState();
+    const [dragOver, setDragOver] = React.useState();
+    const inputRef = useForwardedRef(ref);
+    const controlRef = useRef();
+    const RemoveIcon = theme.fileInput.icons.remove;
+    // When multiple and more than this #, don't show the files individually
+    const aggregateThreshold = 10;
```

Comment:
Should this appear in the documentation?

Reply:
yep, added